### PR TITLE
[CRB-148] Enable coverage

### PR DESCRIPTION
**Pull request by atodorov**

### CODE REVIEW

**Comment by atodorov** [comment]
@nathanwhit please review. The items I'm not confident about are:

1) the change to the nightly toolchain (required by grcov)
2) the actual flags that are specified. I copied these from a readme file, not sure how they affect things on our side.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,27 +29,43 @@ jobs:
...
+      matrix:
+        feature: ["", "--features 'old-sawtooth'"]
+
     steps:
       - uses: actions/checkout@v2
       - name: Install Rust toolchain
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: stable
+          toolchain: nightly
```
Comment (nathanwhit) [inline .github/workflows/ci.yml:43]: Using nightly for the coverage is fine, but I would prefer to run tests on stable as well. The nightly compiler sometimes contains regressions that cause compilation to fail, which also fails the CI job.

The cleanest solution is most likely to split testing into two jobs: one job for running the actual tests - using the stable compiler and without the added flags, and another job for collecting coverage - using the nightly compiler and the added flags. That way, even in the case of a regression in nightly, we'll always get meaningful results from running the test suite.

**Review by nathanwhit — CHANGES_REQUESTED** [review]
Aside from the nightly issue (see comment), the rest of this looks good to me.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (atodorov) [reply .github/workflows/ci.yml]: Should be ready to go. I just added another variable to the test matrix coupled with a conditional to submit the coverage report.

**Review by nathanwhit — CHANGES_REQUESTED** [review]
One last suggestion, then should be good to go.

**Review by nathanwhit — APPROVED** [review]

**Pull request merged**